graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfile.java

```diff
@@ -17,20 +17,18 @@
 package org.graylog2.indexer.indexset.profile;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.graylog2.database.MongoEntity;
 import org.graylog2.indexer.indexset.CustomFieldMappings;
-import org.mongojack.Id;
-import org.mongojack.ObjectId;
 
 import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
 
 /**
  * Represents profile as it is stored in Mongo.
  */
-public record IndexFieldTypeProfile(@JsonProperty(ID_FIELD_NAME) @Nullable @Id @ObjectId String id,
+public record IndexFieldTypeProfile(String id,
```

Comment:
Does annotation inheritance work for records as well? It works for abstract auto-value classes. I'm not sure if it works for records as well. Please check. Thanks!